### PR TITLE
Add docs for upgrade/update

### PR DESCRIPTION
**Pull request by ras0219-msft**
See title.

### CODE REVIEW

File: vcpkg/commands/upgrade.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Synopsis
+
+```console
+vcpkg upgrade [options]
+```
+
+## Description
+
+Remove and reinstalls [Classic mode] packages that have different current available versions.
```
Comment (AugP) [inline vcpkg/commands/upgrade.md:16]: Removes*

File: vcpkg/commands/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+If you are sure you want to rebuild the above packages, run this command with the --no-dry-run option.
+```
+
+## Options
+
+All vcpkg commands support a set of [common options](common-options.md).
+
+### `--no-dry-run`
+
+Perform the planned removals and reinstalls.
```
Comment (AugP) [inline vcpkg/commands/upgrade.md:43]: Suggested rewording: "Removes the ports that are outdated and reinstalls them at the most recent version listed in the user's local copy of the public vcpkg registry".

File: vcpkg/commands/upgrade.md (hunk 1 above)
Comment (AugP) [inline vcpkg/commands/upgrade.md:16]: I think something should be said here that it's more of a preview of what will be changed, but that it only performs the changes when running with --no-dry-run.

**Review by AugP — CHANGES_REQUESTED** [review]

**Review by AugP — APPROVED** [review]

**Comment by Neumann-A** [comment]
Shouldn't there a big warning that these commands are not transactional and will lose the information of which ports were installed if a failure happens.

**Comment by ras0219-msft** [comment]
That big warning is included on the page for `upgrade`:

> [!WARNING]
> This command does not save the previous install state. If an error occurs while building replacement packages, the previous set of packages will not be restored.
>
> For robust version management, use [Manifest mode].

**Pull request merged**